EIPS/eip-7017.md

```diff
@@ -0,0 +1,281 @@
+    /// or msg.sender.
+    event BroadcastMsg (address indexed from, string data);
+    
+    /**
+    * @dev Send a message to an address from the address executing the function
+    * @param to address to send a notification to
+    * @param data to send: (json format, URI recomended): 
+    * {subject, body, image_url (optional),
+    * transaction_request (optional)}
+    */
+    function walletDM (address to, string memory data) external;
+
+    /**
+    * @dev Send a message to an address from the smart contract
+    * @param to address to send a notification to
+    * @param data to send: (json format, URI recomended):
+    * {subject, body, image_url (optional),
+    * transaction_request (optional)}
+    */
+    function contractDM (address to, string memory data) external;
+    
+    /**
+    * @dev Send a general notification from the address executing the function
+    * @param data to broadcast: (json format, URI recomended):
+    * {subject, body, attention_level (1, 2 or 3), image_url (optional),
+    * transaction_request (optional)}
+    */
+    function walletBroadcast (string memory data) external;
+
+    /**
+    * @dev Send a general notification from the address executing the function
+    * @param data to broadcast: (json format, URI recomended):
+    * {subject, body, attention_level (1, 2 or 3), image_url (optional),
+    * transaction_request (optional)}
+    */
+    function contractBroadcast (string memory data) external;
+}
+```
+
+The event's are the message couriers and the functions call the events:
+
+Every `walletBroadcast` and `contractBroadcast` function **MUST** implement the `BroadcastMsg` event with `from` set as `msg.sender` and `address(this)` respectively.
+
+Also, every `walletDM` and `contractDM` function **MUST** implement the `DirectMsg` event with `from` set as `msg.sender` and `address(this)` respectively.
+
+Here is a table to better represent this:
+
+|          | DirectMessage                                     | Broadcast                                        |
+| -------- | ------------------------------------------------- | ------------------------------------------------ |
+| Wallet   | emit DirectMsg(msg.sender, to, message);    | emit BroadcastMsg(msg.sender, message);    |
+| Contract | emit DirectMsg(address(this), to, message); | emit BroadcastMsg(address(this), message); |
+
+### Data Schema
+
+The `data` field **REQUIRED** in the message events **MUST** have a JSON schema that can be written directly as a string, or passed as a URI containing the JSON. It is **RECOMENDED** to use a URI, as it is a cheaper way to send the data.
+
+It has two different specifications, one for broadcast messages and another one for direct messages. Only small differences exist between them. This is the base specifications for both:
+
+```json
+{
+"subject": "The subject of the message",
+"body": "The body of the message",
+"image_uri": "An image URI",
+"transaction_request": "A transaction request"
```

Comment:
it looks like you are missing indentations here. would recommend trying to match lines 151–154, and also clarifying how indentation/spacing should be handled in the JSONs.